### PR TITLE
Enhancements to incorporate CorrectIQ

### PR DESCRIPTION
**Pull request by ghostop14**

### CODE REVIEW

File: grep-9999-correctiq.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+History:
+- 07-Jan-2020: Initial Draft
+
+## Abstract
+
+The DC spike created by the IQ sampling process has long been an issue with the sampling 
+process. Software such as GQRX and SDRSharp have built-in algorithms that can be enabled 
+to remove this artifact.  However, GNU Radio has no built-in blocks to address 
+this feature.  A second related feature common in receiver software is the ability 
+to correct for inverted spectrum.  The gr-correctiq OOT module provides blocks that
```
Comment (mbr0wn) [inline grep-9999-correctiq.md:16]: I don't think this PR justifies a new in-tree component. The blocks, however, are valuable and interesting. They could go into gr-blocks.

File: grep-9999-correctiq.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+to remove this artifact.  However, GNU Radio has no built-in blocks to address 
+this feature.  A second related feature common in receiver software is the ability 
+to correct for inverted spectrum.  The gr-correctiq OOT module provides blocks that 
+address both of these issues.  Therefore, this GREP is to propose that gr-correctiq 
+be merged into GNU Radio core to make these solutions immediately available to users 
+without the need to install the OOT modules.
+
+## Copyright / License
+
+[I could use a suggestion here.  I'd like to retain the right to use the code as I
```
Comment (mbr0wn) [inline grep-9999-correctiq.md:23]: This is the copyright for the GREP itself, not the code that will come out of it. You can copy the line from other GREPs, if you like. It won't affect the code.
Note that the CLA you signed for the code itself will let you retain use of the code (i.e., after it goes into GNU Radio). You don't give up ownership when submitting code, you just can't force GNU Radio to remove your code again in the future.

**Review by mbr0wn — COMMENTED** [review]
@ghostop14 I'm leaning towards merging this PR with a "reject" (i.e. we won't add another in-tree component), but I would really like to see the blocks.

**Comment by ghostop14** [comment]
The blocks are up here: https://github.com/ghostop14/gr-correctiq

I actually consider them core to my builds since the primary block removes the DC spike like other solutions.  Was one of the early issues I had with signals when I first started in that GQRX and SDR# had a simple "checkbox" to make it go away.   These blocks tied in with a checkbox and an IO selector make it easy to put the same functionality in a GR GUI.  Since most folks on say Windows can't build OOT's from source as easily, incorporating it would give them the same core functionality.  Just some more food for thought :)

**Comment by mbr0wn** [comment]
@ghostop14 Not sure we're misunderstanding each other. I really want those blocks, I just don't think they justify an in-tree component.

**Comment by ghostop14** [comment]
I think I'm confusing in-tree with "included in general".  Still getting the hang of this process.  Is that what you meant by going in gr-blocks?  They would go best in there?

**Comment by dkozel** [comment]
@ghostop14 Martin is recommending that these blocks go into gr-blocks and stating that the change here (adding a few blocks similar to existing ones) doesn't need a GREP, so go ahead and open a PR on https://github.com/gnuradio/gnuradio and he/we will take care of closing up the GREP in such a way that it's obvious why it isn't an `accepted` GREP and that the blocks are getting merged.

**Comment by ghostop14** [comment]
Ah gotcha.  Thanks guys!  Bear with me while I'm learning the PR/GREP lingo.

**Comment by ghostop14** [comment]
Looking at this now.  Question... I'm looking through the core block categories in the GUI and trying to see which category fits best.  What do you think, since it's kind of an IIR filter would it go best under filter?

**Comment by mbr0wn** [comment]
Also closing this GREP. As you can tell from #26, we don't really have good rules for what to put where, so I'll have to rely on your judgement and that of the reviewers to discuss where to put stuff.

**Pull request closed without merging**